lib/app/migrate.ts

```diff
@@ -0,0 +1,608 @@
+        text: i18n(
+          `commands.project.subcommands.migrateApp.spinners.migrationFailed`
+        ),
+      });
+    }
+
+    return buildId;
+  } catch (error) {
+    SpinniesManager.fail('finishingMigration', {
+      text: i18n(
+        `commands.project.subcommands.migrateApp.spinners.migrationFailed`
+      ),
+    });
+    throw error;
+  }
+}
+
+export async function downloadProjectFiles(
+  derivedAccountId: number,
+  projectName: string,
+  buildId: number,
+  projectDest: string
+): Promise<void> {
+  try {
+    SpinniesManager.add('fetchingMigratedProject', {
+      text: i18n(
+        `commands.project.subcommands.migrateApp.spinners.downloadingProjectContents`
+      ),
+    });
+
+    const { data: zippedProject } = await downloadProject(
+      derivedAccountId,
+      projectName,
+      buildId
+    );
+
+    const absoluteDestPath = projectDest
+      ? path.resolve(getCwd(), projectDest)
+      : getCwd();
+
+    await extractZipArchive(
+      zippedProject,
+      sanitizeFileName(projectName),
+      absoluteDestPath,
+      {
+        includesRootDir: true,
+        hideLogs: false,
+      }
+    );
+
+    SpinniesManager.succeed('fetchingMigratedProject', {
+      text: i18n(
+        `commands.project.subcommands.migrateApp.spinners.downloadingProjectContentsComplete`
+      ),
+    });
+
+    logger.success(`Saved ${projectName} to ${projectDest}`);
+  } catch (error) {
+    SpinniesManager.fail('fetchingMigratedProject', {
+      text: i18n(
+        `commands.project.subcommands.migrateApp.spinners.downloadingProjectContentsFailed`
+      ),
+    });
+    throw error;
+  }
+}
+
+export async function migrateApp2025_2(
+  derivedAccountId: number,
+  options: ArgumentsCamelCase<MigrateAppOptions>
+): Promise<void> {
+  SpinniesManager.init();
+
+  const { appIdToMigrate, projectName, projectDest } =
+    await handleMigrationSetup(derivedAccountId, options);
+
+  if (!appIdToMigrate || !projectName || !projectDest) {
+    return;
+  }
+
+  const migrationInProgress = await beginMigration(
+    derivedAccountId,
+    appIdToMigrate,
+    options.platformVersion
+  );
+
+  if (!migrationInProgress) {
+    return;
+  }
+
+  const { migrationId, uidMap } = migrationInProgress;
+  const buildId = await finalizeMigration(
+    derivedAccountId,
+    migrationId,
+    uidMap,
+    projectName
+  );
+
+  if (!buildId) {
+    throw new Error('Migration Failed');
+  }
+
+  await downloadProjectFiles(
+    derivedAccountId,
+    projectName,
+    buildId,
+    projectDest
+  );
+}
+
+export function logInvalidAccountError(i18nKey: string): void {
+  uiLine();
+  logger.error(i18n(`${i18nKey}.errors.invalidAccountTypeTitle`));
+  logger.log(
+    i18n(`${i18nKey}.errors.invalidAccountTypeDescription`, {
+      useCommand: uiCommandReference('hs accounts use'),
+      authCommand: uiCommandReference('hs auth'),
+    })
+  );
+  uiLine();
+}
+
+export async function migrateApp2023_2(
```

Comment:
This method is pretty much a direct copy paste

Reply:
Not a blocker, but one alternate option is to move this into a dedicated legacy file just to keep the new logic clean (and make this file a little easier to parse)

Reply:
Yeah, I'll move it.  I had the thought of making it a second file, and it was easier to have them side by side during development, but now the file is pretty large.

Reply:
Moved